addons/website_sale/controllers/main.py

```diff
@@ -265,41 +241,33 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
 
     @http.route(['/shop/product/<model("product.template"):product>'], type='http', auth="public", website=True)
     def product(self, product, category='', search='', **kwargs):
-        cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
-        category_obj = pool['product.public.category']
-        template_obj = pool['product.template']
-
-        context.update(active_id=product.id)
+        local_context = dict(request.env.context, active_id=product.id)
+        ProdcutCategory = request.env['product.public.category']
+        Rating = request.env['rating.rating']
 
         if category:
-            category = category_obj.browse(cr, uid, int(category), context=context)
-            category = category if category.exists() else False
+            category = ProdcutCategory.with_context(local_context).browse(int(category)).exists()
```

Comment:
@rga-odoo why do you use the local_context here? I think the `active_id` is only used in product price but maybe I missed something. Please let me know if I need to keep it or I can remove it.